src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets

```diff
@@ -1,10 +1,16 @@
 <Project>
   <Target Name="_MonoReadAvailableComponentsManifest">
-    <Error Condition="'%(ResolvedRuntimePack.PackageDirectory)' == '' and '$(_MonoRuntimeComponentManifestJsonFilePath)' == ''"
-           Text="Empty ResolvedRuntimePack.PackageDirectory while trying to read runtime components manifest" />
+    <Error Condition="'%(ResolvedFrameworkReference.Identity)' == 'Microsoft.NETCore.App' and '%(RuntimePackPath)' == '' and '$(_MonoRuntimeComponentManifestJsonFilePath)' == ''"
+           Text="Empty ResolveFrameworkReference.RuntimePackPath while trying to read runtime components manifest" />
+
     <PropertyGroup>
-      <_MonoRuntimeComponentManifestJsonFilePath Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == ''">%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\RuntimeComponentManifest.json</_MonoRuntimeComponentManifestJsonFilePath>
+      <_MonoRuntimeComponentManifestJsonFilePath 
+        Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == '' and '%(ResolvedFrameworkReference.Identity)' == 'Microsoft.NETCore.App'">$([MSBuild]::NormalizePath('%(RuntimePackPath)', 'runtimes', '$(RuntimeIdentifier)', 'build', 'RuntimeComponentManifest.json'))</_MonoRuntimeComponentManifestJsonFilePath>
     </PropertyGroup>
+
+    <Error Condition="!Exists('$(_MonoRuntimeComponentManifestJsonFilePath)')"
+           Text="Unable to read the runtime components manifest as the path does not exist" />
```

Comment:
nit: having the path in the error message would be useful.

Reply:
Keeping me honest ;-)